extensions/src/org/pentaho/platform/osgi/KarafPropertyImporter.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright 2015 Pentaho Corporation. All rights reserved.
+ */
+
+package org.pentaho.platform.osgi;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import org.apache.commons.lang.StringUtils;
+
+public class KarafPropertyImporter {
+  final static String delimiter = ",";
+  protected static KarafInstance karafInstance;
+  private String importFilePath;
+  int lineNumber;
+
+  public KarafPropertyImporter( String importFilePath ) {
```

Comment:
I would rather see this rewritten to use RegEx and capture groups within that to extract the values. At least with RegEx I can see the patterns and easily understand how the file should look. All of the split/indexOf/loop code is hard to follow